scripts/SREP/retry-pruning-cronjob/script.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+# Delete jobs that fail in openshift-sre-pruning namespace.
+
+set -x
+set -e
+set -o nounset
+set -o pipefail
+
+readonly PRUNING_NS=openshift-sre-pruning
+
+Builds_Pruner=()
+Deployements_Pruner=()
+FailedJobs=()
+
+Builds_Pruner+=($(oc get jobs -n "$PRUNING_NS" -o=jsonpath='{.items[?(@.metadata.ownerReferences[*].name=="builds-pruner")].metadata.name}'))
+Deployements_Pruner+=($(oc get jobs -n "$PRUNING_NS" -o=jsonpath='{.items[?(@.metadata.ownerReferences[*].name=="deployments-pruner")].metadata.name}'))
+
+detect_job() {
+  echo "INFO: finding jobs producing error"
+  FailedJobs+=($(oc get jobs -n "$PRUNING_NS" -o=jsonpath='{.items[?(@.status.failed>1)].metadata.name}'))
+}
+
+delete_job() {
+  if [[ ${#FailedJobs[@]} == 0 ]]; then
+    echo "INFO: no failed jobs found exiting.."
+    return 1
+  else
+    echo "INFO: deleting jobs producing error"
+    for jobs in "${FailedJobs[@]}"; do
+      oc delete job "$jobs" -n "$PRUNING_NS"
+    done
+  fi
+}
+
+rerun_job() {
+  for jobs in "${FailedJobs[@]}"; do
```

Comment:
If there are multiple failed jobs with the same type, like builds-pruner, I do not think we need to rerun each of them.
We need retry it once.